sdk/apis/apis/v1alpha2/types_apibinding_conversion.go

```diff
@@ -0,0 +1,203 @@
+	// FieldSelector, which went away in v1alpha2.
+	if len(in.Spec.PermissionClaims) > 0 {
+		if out.Annotations == nil {
+			out.Annotations = make(map[string]string)
+		}
+		encoded, err := json.Marshal(in.Spec.PermissionClaims)
+		if err != nil {
+			return err
+		}
+		out.Annotations[PermissionClaimsV1Alpha1Annotation] = string(encoded)
+	}
+
+	if overhangingAPC, ok := in.Annotations[AcceptablePermissionClaimsAnnotation]; ok {
+		acceptablePermissionClaims := []AcceptablePermissionClaim{}
+		if err := json.Unmarshal([]byte(overhangingAPC), &acceptablePermissionClaims); err != nil {
+			return fmt.Errorf("failed to decode claims from JSON: %w", err)
+		}
+
+		for _, pc := range acceptablePermissionClaims {
+			for i, opc := range out.Spec.PermissionClaims {
+				if pc.EqualGRI(opc.PermissionClaim) {
+					out.Spec.PermissionClaims[i].PermissionClaim.Verbs = pc.Verbs
+				}
+			}
+		}
+
+		delete(out.Annotations, AcceptablePermissionClaimsAnnotation)
+
+		// make tests for equality easier to write by turning []string into nil
+		if len(out.Annotations) == 0 {
+			out.Annotations = nil
+		}
+	}
+
+	for i, opc := range out.Spec.PermissionClaims {
+		if len(opc.PermissionClaim.Verbs) == 0 {
+			out.Spec.PermissionClaims[i].PermissionClaim.Verbs = []string{"*"}
+		}
+	}
+
+	return nil
+}
+
+func Convert_v1alpha1_AcceptablePermissionClaims_To_v1alpha2_AcceptablePermissionClaims(in []apisv1alpha1.AcceptablePermissionClaim, s kubeconversion.Scope) (out []AcceptablePermissionClaim, overhanging []apisv1alpha1.AcceptablePermissionClaim, err error) {
+	for _, apc := range in {
+		// ResourceSelector has been removed from PermissionClaims, we can't cleanly convert it.
+		if apc.PermissionClaim.All && len(apc.PermissionClaim.ResourceSelector) == 0 {
+			var v2apc AcceptablePermissionClaim
+			if err := Convert_v1alpha1_AcceptablePermissionClaim_To_v1alpha2_AcceptablePermissionClaim(&apc, &v2apc, s); err != nil {
+				return nil, nil, err
+			}
+			out = append(out, v2apc)
+		} else {
+			overhanging = append(overhanging, apc)
+		}
+	}
+
+	return
+}
+
+func Convert_v1alpha1_AcceptablePermissionClaim_To_v1alpha2_AcceptablePermissionClaim(in *apisv1alpha1.AcceptablePermissionClaim, out *AcceptablePermissionClaim, s kubeconversion.Scope) error {
+	if err := Convert_v1alpha1_PermissionClaim_To_v1alpha2_PermissionClaim(&in.PermissionClaim, &out.PermissionClaim, s); err != nil {
+		return err
+	}
+
+	out.State = AcceptablePermissionClaimState(in.State)
+	out.Selector.MatchAll = in.PermissionClaim.All
+	if len(out.PermissionClaim.Verbs) == 0 {
+		out.PermissionClaim.Verbs = []string{"*"}
+	}
+
+	return nil
+}
+
+func Convert_v1alpha1_PermissionClaim_To_v1alpha2_ScopedPermissionClaim(in *apisv1alpha1.PermissionClaim, out *ScopedPermissionClaim, s kubeconversion.Scope) error {
+	return Convert_v1alpha1_PermissionClaim_To_v1alpha2_PermissionClaim(in, &out.PermissionClaim, s)
```

Comment:
Shouldn't we also set `MatchAll` in this conversion given its available?

Reply:
Updated.